File: shopsphere-frontend/src/services/addressService.js
import axiosInstance from "./axiosInstance";

export const getAddresses = () => {
  return axiosInstance.get("/address");
};

export const addAddress = (addressData) => {
  return axiosInstance.post(
    "/address",
    addressData
  );
};

export const updateAddress = (
  addressId,
  addressData
) => {
  return axiosInstance.put(
    `/address/${addressId}`,
    addressData
  );
};

export const deleteAddress = (
  addressId
) => {
  return axiosInstance.delete(
    `/address/${addressId}`
  );
};

File: Backend/src/main/java/com/shopsphere/backend/service/AddressService.java
package com.shopsphere.backend.service;

import com.shopsphere.backend.dto.address.AddressRequest;
import com.shopsphere.backend.dto.address.AddressResponse;

import java.util.List;

public interface AddressService {

    AddressResponse addAddress(AddressRequest request);

    List<AddressResponse> getMyAddresses();

    AddressResponse updateAddress(Long addressId, AddressRequest request);

    void deleteAddress(Long addressId);
}